pkg/morph/client/notary.go

```diff
@@ -296,6 +296,51 @@ func (c *Client) NotaryInvokeNotAlpha(contract util.Uint160, fee fixedn.Fixed8,
 	return c.notaryInvoke(false, false, contract, method, args...)
 }
 
+// NotarySignAndInvokeTX signs and sends notary request that was received from
+// Notary service.
+// NOTE: does not fallback to simple `Invoke()`. Expected to be used only for
+// TXs received from notary requests.
+func (c *Client) NotarySignAndInvokeTX(mainTx *transaction.Transaction) error {
+	if c.multiClient != nil {
+		return c.multiClient.iterateClients(func(c *Client) error {
+			return c.NotarySignAndInvokeTX(mainTx)
+		})
+	}
+
+	alphabetList, err := c.notary.alphabetSource()
+	if err != nil {
+		return fmt.Errorf("could not fetch current alphabet keys: %w", err)
+	}
+
+	multiaddrAccount, err := c.notaryMultisigAccount(alphabetList, false, true)
+	if err != nil {
+		return err
+	}
+
+	// mainTX is expected to be pre-validated: second witness must exist and be empty
+	mainTx.Scripts[1].VerificationScript = multiaddrAccount.GetVerificationScript()
+	mainTx.Scripts[1].InvocationScript = append(
+		[]byte{byte(opcode.PUSHDATA1), 64},
+		multiaddrAccount.PrivateKey().SignHashable(uint32(c.client.GetNetwork()), mainTx)...,
+	)
+
+	resp, err := c.client.SignAndPushP2PNotaryRequest(mainTx,
+		[]byte{byte(opcode.RET)},
+		-1,
+		0,
+		c.notary.fallbackTime,
```

Comment:
To provide the same `NotValidBefore` value for all alphabet fallbacks, you can use `NotValidBefore` value of the original (i.e. sent by storage node) fallback transaction. Otherwise current `NotValidBefore` alphabet fallback values may slightly differ.

Reply:
do you mean to pass `fallbackTx.ValidUntilBlock - fallbackTx.Attributes[NotValidBeforeT] + 1` as 5th arg here?

Reply:
(`fallbackTx.Attributes[NotValidBeforeT]` is shorthand)

Reply:
I just realised, that it is already the proper one, because `c.notary.fallbackTime` is a constant and fallback's `ValidUntilBlock` equals to the main's one, so it's OK to keep the existing code.